docs/Parser.puml

```diff
@@ -1,37 +1,53 @@
 @startuml
-abstract class Parser {
-    commandFormat: String
-    groupNames: HashSet<String>
-    parsedCommand: HashMap<String, String>
-    --
-    + parseString(String str): HashMap<String, String>
-}
 
-class XYZParser {
-    + parseCommand(String userInp): Command
+package parsers {
+    abstract class Parser {
+        commandFormat: String
+        groupNames: HashSet<String>
+        parsedCommand: HashMap<String, String>
+        --
+        + parseString(String str): HashMap<String, String>
+    }
+
+    class XYZParser {
+        + parseCommand(String userInput): Command
+    }
+
+    class ModHappyParser {
+        - getCommandParser(String commandWord): Parser
+        + parseCommand(String userInput): Command
+    }
 }
 
+note top of XYZParser
+    XYZParser is any command-specific
+    parser (e.g. AddParser, TagParser)
+end note
+
+Parser <|-u- ModHappyParser
+XYZParser <. ModHappyParser: uses <
+ModHappyParser <.u. Main
```

Comment:
Should here be navigability instead because Main holds a reference of ModHappyParser?